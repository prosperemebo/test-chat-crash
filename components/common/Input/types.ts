import { TextInputProps } from 'react-native'

interface Props {
    error?: string
    iconName?: 'envelope' | 'unlocked' | 'key'
    type: 'email' | 'password' | 'general' | 'username'
    value: string,
    placeholder: string,
}

export type InputProps = Props & TextInputProps