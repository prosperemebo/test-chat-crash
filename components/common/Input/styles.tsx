import React from "react"
import styled, { css } from "styled-components/native"
import { StyleSheet, TextInput } from "react-native"
import FontAwesome from '@expo/vector-icons/FontAwesome5'
import Fontisto from '@expo/vector-icons/Fontisto'

type StyledIconProps = {
  isError: boolean
}

type ContainerProps = {
  showError: boolean
}

export const Container = styled.View<ContainerProps>`
  flex-direction: row;
  align-items: center;
  padding-horizontal: 10px;
  border-radius: 8px;
  border-width: ${StyleSheet.hairlineWidth};
  border-color: ${(props) => props.theme.common.inputBorderColor};
  background-color: ${(props) => props.theme.common.inputBackgroundColor};
  ${props => {
  if (props.showError)
    return css`
        border-color: ${(props) => props.theme.common.inputErrorBorderColor};
      `
}}
  padding-vertical: 10px;
`

export const StyledIcon = styled(FontAwesome)<StyledIconProps>`
    font-size: 20px;
    color: ${props => (props.isError ? props.theme.common.inputTextError : props.theme.common.inputIconColor)};
    margin-right: 10px;
`

export const StyledIconN = styled(Fontisto)<StyledIconProps>`
    font-size: 20px;
    color: ${props => (props.isError ? props.theme.common.inputTextError : props.theme.common.inputIconColor)};
    margin-right: 10px;
`

export const StyledTextInput = styled(TextInput)`
    color: ${props => props.theme.common.inputTextColor};
    flex: 1;
    padding: 0;
    font-size: 14px;
    height: 32px;
    font-family: 'Poppins-Regular';
`

export const ErrorLabel = styled.Text`
  color: ${props => props.theme.common.inputTextError};
  font-size: 14px;
  font-family: 'Poppins-Regular';
  position: relative;
  margin-bottom: 5px;
`

export const ToggleButton = styled.TouchableOpacity``