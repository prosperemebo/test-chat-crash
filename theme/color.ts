export const WHITE = '#FFFFFF'
export const t10WHITE = 'rgba(255,255,255,0.1)'
export const t76WHITE = 'rgba(255,255,255,0.76)'

export const RED = '#f60004'
export const RED1 = '#D33E41'


export const BLACK = '#000000'
export const BLACK2 = '#050608'
export const nBLACK = '#15161A'
export const nBLACK10 = '#1F2025'
export const nBLACK30 = '#28292E'
export const nBLACK40 = '#050608'

export const nGRAY10 = '#6B6E79'
export const nGRAY20 = '#858995'
export const nGRAY25 = '#54575e'
export const nGRAY30 = '#2f3137'
export const nGRAY35 = '#52555D'
export const nGRAY40 = '#35363B'


export const tGRAY20 = '#85899599'
export const tGRAY30 = '#545458A6'
export const tGRAY40 = '#45464C'
export const GRAY40 = '#cfcfd0'
export const GRAY45 = '#c9c9cb'
export const GRAY50 = '#aaaaaa'
export const GRAY60 = '#52545D'
export const t40GRAY60 = '#52545D66'
export const GRAY70 = '#383B40'
export const GRAY90 = '#27292d'
export const GRAY100 = '#1f2023'
export const GRAY110 = '#151619'

export const GRAY_ONBOARD = '#8791a5'

export const BLUE = '#50b5e8'
export const BLUE2 = '#0A84FF'

export const PURPLE = '#9A1F99'

export const nBLUE = '#8D88A8'
export const nPURPLE = '#C47AF5'
export const nPURPLE10 = '#873FBB'
export const nPURPLE20 = '#873FBB'
export const nPURPLE40 = '#C882F8'

export const nORANGE10 = '#FF6705'
export const nORANGE20 = '#FF7905'
export const nORANGE40 = '#FFB700'

export const YELLOW2 = '#EBC789'
export const nYELLOW = '#F5D193'

export const nGRAYBACKGROUND2 = '#292A30'
export const nGRAYBACKGROUND3 = '#383940'
export const nGRAYBACKGROUND4 = '#989898'

export const gradient = {
  COLORS: [BLUE, PURPLE],
  START: { x: 0, y: 1 },
  END: { x: 1, y: 0 },
  LABEL_COLOR: WHITE,
}

export const orangeGradient = {
  COLORS: [nORANGE10, nORANGE40],
  START: { x: 0, y: 1 },
  END: { x: 1, y: 0 },
  LABEL_COLOR: WHITE,
}