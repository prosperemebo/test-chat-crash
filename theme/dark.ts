import {
  WHITE,
  BLACK,
  nGRAY10,
  GRAY_ONBOARD,
  nPURPLE,
  nBLUE,
  GRAY40,
  nGRAY20,
  RED1,
  tGRAY20,
  RED,
  nBLACK,
  nORANGE10,
  nORANGE40,
  nPURPLE10,
  GRAY70,
  tGRAY30,
  BLUE2,
  BLACK2,
  nGRAY25,
  nBLACK30,
  GRAY50,
  nBLACK10,
  t10WHITE,
  nGRAYBACKGROUND2,
  GRAY100,
  GRAY60,
  nGRAY40,
  nGRAY30,
  YELLOW2,
  nYELLOW,
  nGRAYBACKGROUND3,
  tGRAY40,
  nGRAY35,
  nBLACK40,
  t40GRAY60,
  nPURPLE20,
  nPURPLE40,
} from './color'

export const dark = {
  tint: nPURPLE,
  primaryTextColor: WHITE,
  primaryBg: nBLACK10,
  headerBg: GRAY100,
  headerBorderColor: GRAY60,
  secondaryTextColor: GRAY60,
  secondaryBgLight: nBLACK30,
  secondaryBg: nBLACK,
  serviceBg: GRAY100,
  inActiveBorderColor2: t10WHITE,
  backgroundColor2: nGRAYBACKGROUND2,
  backgroundColor3: nGRAYBACKGROUND3,
  backgroundColor4: nBLACK10,

  auth: {
    socialIconColor: WHITE,
    socialLoginButton: BLACK,
    dividerColor: nGRAY10,
    noticeText: WHITE,
    forgotPasswordLabel: WHITE,
  },

  onboarding: {
    headerTextColor: WHITE,
    headerStepsTextColor: GRAY_ONBOARD,
    progressBarColor: nPURPLE,
    progressBarBackgroundColor: nBLUE,
    selectedItemTextColor: WHITE,
    disabledItemTextColor: GRAY40,
    gameSectionOverlayColor: nPURPLE,
    gameEmptySectionColor: BLACK,
    gameEmptyIconColor: BLACK2,
  },

  common: {
    labelColor: WHITE,
    infoTextColor: nGRAY20,
    inputIconColor: nGRAY20,
    inputPlaceholderColor: nGRAY20,
    serviceTextColor: WHITE,
    inputTextColor: WHITE,
    inputTextError: RED1,
    inputBorderColor: tGRAY20,
    inputErrorBorderColor: RED,
    inputBackgroundColor: nBLACK,
    toastBackground: nBLACK,
    toastBorderColor: nPURPLE,
    toastTextColor: WHITE,
    buttonGradientOrangeStart: nORANGE10,
    buttonGradientOrangeEnd: nORANGE40,
    buttonGradientPurpleStart: nPURPLE40,
    buttonGradientPurpleEnd: nPURPLE20,
    buttonGradientGrayStart: nGRAY10,
    buttonGradientGrayEnd: nGRAY20,
    buttonDartBackgroundColor: BLACK,
    backButtonColor: nGRAY20,
    switchTextColor: WHITE,
    checkboxIconColor: nPURPLE,
    gradientButtonColorStart: nPURPLE10,
    gradientButtonColorEnd: nPURPLE,
    borderColor: GRAY70,
    genderPickerBorderColor: nGRAY25,
    genderPickerActiveTextColor: WHITE,
    genderPickerTextColor: GRAY50,
    checkboxTextColor: WHITE,
    iconButtonColor: WHITE,
    backgroundColor: nBLACK10,
  },

  settings: {
    socialIconBackgroundColor: nGRAY40,
    premiumFeatureTextColor: nPURPLE,
    buttonIconColor: nGRAY20,
    serviceItemBgColor: nGRAY30,
  },

  postOnboarding: {
    backgroundColor: nBLACK10,
    inputIconColor: nGRAY20,
    inputIconErrorColor: RED1,
    disclosureTextColor: nGRAY20,
    loadingIndicatorColor: BLACK,
  },

  profile: {
    backgroundColor: nBLACK10,
    improveBackgroundColor: YELLOW2,
    improveBannerBackgroundColor: nYELLOW,
    improveBannerTitleColor: nBLACK30,
    coverPictureCardRemoveButtonBg: tGRAY30,
    profilePicturesCountColor: tGRAY40,
  },

  chats: {
    iconColor: nGRAY20,
    inputColor: nBLACK40,
    placeholderColor: nGRAY35,
    appbarBorderColor: t40GRAY60,
    backgroundColor: nBLACK10,
  },

  nav: {
    bottomTabLabelBackgroundColor: nBLACK,
    bottomTabBackgroundColor: nBLACK,
  },

  other: {
    updateApp: {
      textBorderColor: tGRAY30,
      buttonTextColor: BLUE2,
      textColor: WHITE,
    },
  },
}
