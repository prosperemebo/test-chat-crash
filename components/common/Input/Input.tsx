import { useState } from 'react'
import { useTheme } from 'styled-components/native'
import { Container, StyledIcon, StyledIconN, StyledTextInput, ErrorLabel, ToggleButton } from './styles'
import { InputProps } from './types'
import { TextInputProps } from 'react-native'


const Input = (props: InputProps) => {
  const { error, iconName, type = 'general', value, onChangeText, placeholder, ...other } = props
  const theme = useTheme()
  const [showPassword, setShowPassword] = useState(false)

  const togglePassword = () => {
    setShowPassword(!showPassword)
  }

  const renderIcon = () => {
    switch (iconName) {
      case 'key':
      case 'envelope': {
        return <StyledIcon name={iconName} isError={!!error} />
      }
      case 'unlocked': {
        return <StyledIconN name={iconName} isError={!!error} />
      }
      default: return null
    }
  }

  const getExtraProps = (): TextInputProps => {
    if (type === 'email') {
      return {
        textContentType: 'emailAddress',
        keyboardType: 'email-address',
        autoCapitalize: 'none',
        autoCorrect: false,
      }
    }

    if (type === 'password') {
      return {
        secureTextEntry: !showPassword,
        textContentType: 'password',
        autoCapitalize: 'none',
        autoCorrect: false,
      }
    }

    if (type === 'username') {
      return {
        autoCapitalize: 'none',
        autoCorrect: false,
        underlineColorAndroid: 'rgba(0, 0, 0, 0)',
        maxLength: 50,
      }
    }

    return {}
  }

  return <>
    <Container showError={!!error}>
      {renderIcon()}
      <StyledTextInput
        value={value}
        onChangeText={onChangeText}
        placeholder={placeholder}
        underlineColorAndroid='transparent'
        placeholderTextColor={theme.common.inputPlaceholderColor}
        {...getExtraProps()}
        {...other}
      />
      {type === 'password' && <ToggleButton onPress={togglePassword}>
        <StyledIcon name={showPassword ? 'eye' : 'eye-slash'} isError={false} />
      </ToggleButton>
      }
    </Container>
    {error && <ErrorLabel>{error}</ErrorLabel>}
  </>
}

export default Input