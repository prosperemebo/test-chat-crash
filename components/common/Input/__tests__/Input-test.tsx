import { render } from '@testing-library/react-native'
import { ThemeProvider } from 'styled-components/native'
import { describe, expect, test } from '@jest/globals'
import Input from '../'
import { dark } from '@/theme'

describe('<Input />', () => {

  test('Input renders correctly', () => {
    const tree = render(
      <ThemeProvider theme={dark}>
        <Input type='general' value={'234'} placeholder={'Test test'} iconName={'key'}/>
        <Input type='general' value={'234'} placeholder={'Test test'} error={'34758dshfgjkbsdjbsd'}/>
        <Input type='general' value={'234'} placeholder={'Test test'}/>
        <Input type='username' value={'234'} placeholder={'Test test'}/>
        <Input type='email' value={'234'} placeholder={'Test test'}/>
        <Input type='password' value={'234'} placeholder={'Test test'}/>
      </ThemeProvider>,
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})