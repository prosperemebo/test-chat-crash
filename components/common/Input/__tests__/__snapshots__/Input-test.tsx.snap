// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> Input renders correctly 1`] = `
[
  <View
    showError={false}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#15161A",
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "borderColor": "#85899599",
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
        "borderWidth": 0.5,
        "flexDirection": "row",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
  >
    <Text
      allowFontScaling={false}
      isError={false}
      selectable={false}
      style={
        [
          {
            "color": undefined,
            "fontSize": 12,
          },
          {
            "color": "#858995",
            "fontSize": 20,
            "marginRight": 10,
          },
          {
            "fontFamily": undefined,
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {
            "fontWeight": "700",
          },
        ]
      }
    >
      
    </Text>
    <TextInput
      placeholder="Test test"
      placeholderTextColor="#858995"
      style={
        {
          "color": "#FFFFFF",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "fontFamily": "Poppins-Regular",
          "fontSize": 14,
          "height": 32,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        }
      }
      underlineColorAndroid="transparent"
      value="234"
    />
  </View>,
  <View
    showError={true}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#15161A",
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "borderColor": "#f60004",
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
        "borderWidth": 0.5,
        "flexDirection": "row",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
  >
    <TextInput
      placeholder="Test test"
      placeholderTextColor="#858995"
      style={
        {
          "color": "#FFFFFF",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "fontFamily": "Poppins-Regular",
          "fontSize": 14,
          "height": 32,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        }
      }
      underlineColorAndroid="transparent"
      value="234"
    />
  </View>,
  <Text
    style={
      {
        "color": "#D33E41",
        "fontFamily": "Poppins-Regular",
        "fontSize": 14,
        "marginBottom": 5,
        "position": "relative",
      }
    }
  >
    34758dshfgjkbsdjbsd
  </Text>,
  <View
    showError={false}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#15161A",
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "borderColor": "#85899599",
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
        "borderWidth": 0.5,
        "flexDirection": "row",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
  >
    <TextInput
      placeholder="Test test"
      placeholderTextColor="#858995"
      style={
        {
          "color": "#FFFFFF",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "fontFamily": "Poppins-Regular",
          "fontSize": 14,
          "height": 32,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        }
      }
      underlineColorAndroid="transparent"
      value="234"
    />
  </View>,
  <View
    showError={false}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#15161A",
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "borderColor": "#85899599",
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
        "borderWidth": 0.5,
        "flexDirection": "row",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
  >
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      maxLength={50}
      placeholder="Test test"
      placeholderTextColor="#858995"
      style={
        {
          "color": "#FFFFFF",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "fontFamily": "Poppins-Regular",
          "fontSize": 14,
          "height": 32,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        }
      }
      underlineColorAndroid="rgba(0, 0, 0, 0)"
      value="234"
    />
  </View>,
  <View
    showError={false}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#15161A",
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "borderColor": "#85899599",
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
        "borderWidth": 0.5,
        "flexDirection": "row",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
  >
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      keyboardType="email-address"
      placeholder="Test test"
      placeholderTextColor="#858995"
      style={
        {
          "color": "#FFFFFF",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "fontFamily": "Poppins-Regular",
          "fontSize": 14,
          "height": 32,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        }
      }
      textContentType="emailAddress"
      underlineColorAndroid="transparent"
      value="234"
    />
  </View>,
  <View
    showError={false}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#15161A",
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "borderColor": "#85899599",
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
        "borderWidth": 0.5,
        "flexDirection": "row",
        "paddingHorizontal": 10,
        "paddingVertical": 10,
      }
    }
  >
    <TextInput
      autoCapitalize="none"
      autoCorrect={false}
      placeholder="Test test"
      placeholderTextColor="#858995"
      secureTextEntry={true}
      style={
        {
          "color": "#FFFFFF",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "fontFamily": "Poppins-Regular",
          "fontSize": 14,
          "height": 32,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        }
      }
      textContentType="password"
      underlineColorAndroid="transparent"
      value="234"
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        isError={false}
        selectable={false}
        style={
          [
            {
              "color": undefined,
              "fontSize": 12,
            },
            {
              "color": "#858995",
              "fontSize": 20,
              "marginRight": 10,
            },
            {
              "fontFamily": undefined,
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {
              "fontWeight": "400",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>,
]
`;
